src/Xamarin.Android.Build.Tasks/Tasks/Aapt.cs

```diff
@@ -428,16 +428,177 @@ protected void LogEventsFromTextOutput (string singleLine, MessageImportance mes
 					message = message.Substring ("error: ".Length);
 
 				if (level.Contains ("error") || (line != 0 && !string.IsNullOrEmpty (file))) {
-					LogCodedError ("APT0000", message, file, line);
+					LogCodedError (GetErrorCode (message), message, file, line);
 					return;
 				}
 			}
 
 			if (!apptResult) {
-				LogCodedError ("APT0000", string.Format ("{0} \"{1}\".", singleLine.Trim (), singleLine.Substring (singleLine.LastIndexOfAny (new char [] { '\\', '/' }) + 1)), ToolName);
+				var message = string.Format ("{0} \"{1}\".", singleLine.Trim (), singleLine.Substring (singleLine.LastIndexOfAny (new char [] { '\\', '/' }) + 1));
+				LogCodedError (GetErrorCode (message), message, ToolName);
 			} else {
-				LogCodedWarning ("APT0000", singleLine);
+				LogCodedWarning (GetErrorCode (singleLine), singleLine);
 			}
 		}
+
+		static string GetErrorCode (string message)
+		{
+			foreach (var tuple in error_codes)
+				if (message.IndexOf (tuple.Item1, StringComparison.OrdinalIgnoreCase) >= 0)
+					return tuple.Item2;
+
+			return "APT0000";
+		}
+
+		static readonly List<Tuple<string, string>> error_codes = new List<Tuple<string, string>> () {
+			Tuple.Create ("AndroidManifest.xml is corrupt", "APT1100"),
```

Comment:
For ease of human parsing, could the error number be *first* in the tuple, instead of second?  That would allow all of the error messages and codes to line up vertically and be easier (for me) to read:

```csharp
Tuple.Create ("APT1100", "AndroidManifest.xml is corrupt"),
Tuple.Create ("APT1001", "can't use '-u' with add"),
Tuple.Create ("APT1002", "dump failed because assets could not be loaded"),
// ...
```

Reply:
Commit message/PR description should also provide where/how you got these strings.  Grepping the `aapt` source (with what `grep` command)?  Something else?

Reply:
Updated.